client/Wayland/wlfreerdp.c

```diff
@@ -77,26 +81,35 @@ void wl_end_paint(rdpContext* context)
 	wlf_RefreshDisplay(display);
 }
 
-BOOL wl_pre_connect(freerdp* instance)
+static BOOL wl_pre_connect(freerdp* instance)
 {
 	wlfDisplay* display;
 	wlfInput* input;
 	wlfContext* context;
 
-	freerdp_channels_pre_connect(instance->context->channels, instance);
+	if (freerdp_channels_pre_connect(instance->context->channels, instance))
```

Comment:
@hardening that should be fine. freerdp_channels_pre_connect isn't bool it return 0 on success.